mshadow/dot_engine-inl.h

```diff
@@ -400,40 +405,45 @@ struct BLASEngine<cpu, double> {
                                   double beta, double *C, int ldc, int batch_count,
                                   double **workspace) {
 #if (MSHADOW_USE_MKL && INTEL_MKL_VERSION >= 20160000)
-  std::vector<int> p_m(batch_count, m);
-  std::vector<int> p_n(batch_count, n);
-  std::vector<int> p_k(batch_count, k);
-  std::vector<int> p_lda(batch_count, lda);
-  std::vector<int> p_ldb(batch_count, ldb);
-  std::vector<int> p_ldc(batch_count, ldc);
-  std::vector<double> p_alpha(batch_count, alpha);
-  std::vector<double> p_beta(batch_count, beta);
-  std::vector<const double*> pp_A;
-  std::vector<const double*> pp_B;
-  std::vector<double*> pp_C;
+  // since same m/n/k is used for all single gemms, so we put all gemms into one group
+  // group size = 1
```

Comment:
group count = 1,  group size = batch_count,   so p_m, p_n   need to use MKL_INT p_m[batch_count]  that is std::vector<int> p_k(batch_count, k);